src/Cli/OptionCommand.php

```diff
@@ -0,0 +1,264 @@
+			WP_CLI::halt( 1 );
+		}
+
+		if ( ! $this->is_valid_value( $key, $value ) ) {
+			WP_CLI::error( "Could not update '{$key}' option. Is the value valid?", false );
+			WP_CLI::line( "Allowed values for '{$key}' option are: " );
+
+			$allowed_args = $this->get_allowed_args()[ $key ]['enum'];
+
+			/**
+			 * Format array to be used with WP_CLI\Utils\format_items().
+			 */
+			$allowed_args = array_map(
+				static function ( $allowed_arg ) {
+					return [ 'option_value' => $allowed_arg ];
+				},
+				$allowed_args
+			);
+
+			WP_CLI\Utils\format_items( 'table', $allowed_args, [ 'option_value' ] );
+			WP_CLI::halt( 1 );
+		}
+
+		if ( ! AMP_Options_Manager::update_option( $key, $value ) ) {
+			WP_CLI::error( "Could not update '{$key}' option." );
+		}
+
+		WP_CLI::success( "Updated '{$key}' option." );
+	}
+
+	/**
+	 * List AMP options.
+	 *
+	 * @subcommand list
+	 */
+	public function list_() {
+		$amp_options = AMP_Options_Manager::get_options();
+
+		/**
+		 * Format array to be used with WP_CLI\Utils\format_items().
+		 */
+		$amp_options = array_map(
+			static function ( $value, $key ) {
+				return [
+					'option_name'  => $key,
+					'option_value' => $value,
+				];
+			},
+			$amp_options,
+			array_keys( $amp_options )
+		);
+
+		WP_CLI::line( 'AMP options:' );
+		WP_CLI\Utils\format_items( 'table', $amp_options, [ 'option_name', 'option_value' ] );
+
+		WP_CLI::line(); // Add a new line for readability.
+		WP_CLI::line( 'Allowed AMP options to be updated using WP CLI are: ' );
+
+		$allowed_args = $this->get_allowed_args();
+
+		/**
+		 * Format array to be used with WP_CLI\Utils\format_items().
+		 */
+		$allowed_args = array_map(
+			static function ( $allowed_arg, $key ) {
+				return [
+					'option_name'  => $key,
+					'option_value' => implode( ', ', $allowed_arg['enum'] ),
+				];
+			},
+			$allowed_args,
+			array_keys( $allowed_args )
+		);
+
+		WP_CLI\Utils\format_items( 'table', $allowed_args, [ 'option_name', 'option_value' ] );
+	}
+
+	/**
+	 * Construct schema for the options.
+	 *
+	 * @return array
+	 */
+	private function get_allowed_args() {
+		if ( ! $this->allowed_args ) {
+			$this->allowed_args = [
+				Option::THEME_SUPPORT      => [
+					'type' => 'string',
+					'enum' => [
+						AMP_Theme_Support::READER_MODE_SLUG,
+						AMP_Theme_Support::STANDARD_MODE_SLUG,
+						AMP_Theme_Support::TRANSITIONAL_MODE_SLUG,
+					],
+				],
+				Option::MOBILE_REDIRECT    => [
+					'type' => 'boolean',
+					'enum' => [
+						'true',
+						'false',
+					],
+				],
+				Option::READER_THEME       => [
+					'type' => 'string',
+					'enum' => wp_list_pluck( $this->reader_themes->get_themes(), 'slug' ),
+				],
+				Option::SUPPRESSED_PLUGINS => [
+					'type'  => 'array',
+					'items' => [
+						'type' => 'string',
+					],
+					'enum'  => array_keys( $this->plugin_suppression->get_suppressible_plugins_with_details() ),
+				],
+			];
+		}
+
+		return $this->allowed_args;
+	}
```

Comment:
We can of course do internal REST API calls to the endpoints, avoiding the need to do any HTTP requests.

Reply:
I think the user will need to set up `WP User` by appending the global `--user` flag while running the commands if we use the REST API calls for this reason. 

https://github.com/ampproject/amp-wp/blob/5fc17b69954468cb3e65b58f27d4c565fa546f41/src/OptionsRESTController.php#L150

Reply:
Or that line could be modified to allow WP-CLI:

```diff
- if ( ! current_user_can( 'manage_options' ) ) {
+ if ( ! current_user_can( 'manage_options' ) && ! ( defined( 'WP_CLI' ) && WP_CLI ) ) {
```

Although that makes me a bit wary. Should make sure that this is only ever defined in a CLI context.

Reply:
> Or that line could be modified to allow WP-CLI:
> 
> ```diff
> - if ( ! current_user_can( 'manage_options' ) ) {
> + if ( ! current_user_can( 'manage_options' ) && ! ( defined( 'WP_CLI' ) && WP_CLI ) ) {
> ```
> 
> Although that makes me a bit wary. Should make sure that this is only ever defined in a CLI context.

I too explored this possibility but It can cause a security issue if somehow `WP_CLI` is true outside the CLI context.